libbeat/monitoring/monitoring.go

```diff
@@ -89,13 +89,12 @@ func Clear() error {
 // use xpack.monitoring.* settings OR monitoring.* settings but not both.
 func SelectConfig(beatCfg BeatConfig) (*common.Config, *report.Settings, error) {
 	switch {
-	case beatCfg.Monitoring.Enabled() && beatCfg.XPackMonitoring.Enabled():
-		return nil, nil, errMonitoringBothConfigEnabled
```

Comment:
I think we still want to keep this this check around to prevent users from trying to set `xpack.monitoring.*` and `monitoring.*` simultaneously. We probably want to update the error message though, saying something about not using both settings but instead using Metricbeat with the `beat-xpack` module.

Reply:
Can do. I was on the fence about this one but wanted to see what you thought. Thanks!